File: Python/ReverseStringII/ReverseStringII.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        ans = ''
        for i in xrange(0, len(s), k):
            end = min(i + k, len(s))
            ans += s[i:end][::-1] if (i / k) % 2== 0 else s[i:end]
        return ans

File: JavaScript/ReverseStringII/ReverseStringII.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
 //(105ms)
var reverseStr = function(s, k) {
    var ans = '';
    for(let i = 0; i < s.length; i += k) {
        let end = Math.min(i + k, s.length);
        if(Math.floor(i / k) % 2 === 0)
            ans += s.slice(i, end).split('').reverse().join('');
        else
            ans += s.slice(i, end);
    }
    return ans;
};